keras_nlp/src/tests/test_case.py

```diff
@@ -604,5 +604,27 @@ def compare(actual, expected):
 
             tree.map_structure(compare, output, expected_partial_output)
 
+    def run_pyramid_output_test(
+        self,
+        cls,
+        init_kwargs,
+        input_data,
+        expected_pyramid_output_keys,
+    ):
+        """Run Tests for Feature Pyramid output keys and shape."""
+        backbone = cls(**init_kwargs)
+        model = keras.models.Model(backbone.inputs, backbone.pyramid_outputs)
+        output_data = model(input_data)
+
+        self.assertIsInstance(output_data, dict)
+        self.assertEqual(
+            list(output_data.keys()), list(backbone.pyramid_outputs.keys())
+        )
+        self.assertEqual(list(output_data.keys()), expected_pyramid_output_keys)
+
+        for k, v in output_data.items():
+            size = input_data.shape[1] // (2 ** int(k[1:]))
```

Comment:
Is this part true for all feature pyramid outputs? Or only some?

Reply:
This is true for Resnet, Densenet, CSPdarknet. 
For Mix transformer it will be with the slight change `size = input_data.shape[1] // (2 ** (int(k[1:])+1))` which does not fit into this testcase.

Reply:
And When I checked, this holds good for all efficientNet networks, mobilenet as well.

Reply:
Got it. Why don't we write this in a way that would be really general and obvious to the reader. Something like this in the actual test case. Then here we can just assert that `shape[1:3]` is the expected image size.

```
expected_pyramid_output_keys=["P2", "P3", "P4", "P5"],
expected_pyramid_image_sizes=[(128, 128), (64, 64), (32, 32), (16, 16)],
```

Reply:
Changed, Thanks